RHDEVDOCS-4485 migration guide - odo v2 to v3 (#51479)
openshift/openshift-docs: modules/odo-push-watch-to-odo-dev.adoc

// Module included in the following assemblies:
//
// * cli_reference/developer_cli_odo/odo-migration-guide.adoc

:_content-type: CONCEPT
[id="odo-push-watch-to-odo-dev_{context}"]

> = odo push and odo watch replaced by odo dev

Reviewer: ```suggestion
= `odo push` and `odo watch` replaced by `odo dev`
```

`odo` v3 replaces the commands `odo push` and `odo watch` from v2 with a single command `odo dev`. 

> With `odo` v2, if you wanted to automatically synchronize the code running on the developer system with the application running on a Kubernetes cluster, you had to perform two steps:

Reviewer: ```suggestion
In `odo` v2, if you wanted to automatically synchronize the code running on the developer system with the one running on your cluster, you had to perform two steps:
```

I am not sure if it should be 'Kubernetes cluster' or 'OpenShift cluster' since this is a downstream doc, I am more inclined towards the latter.

. `odo push` to start the application on the cluster
. `odo watch` to automatically synchronize the code

To delete the inner loop resources of the component from the cluster, you had to use `odo delete` or`odo component delete`.

With `odo` v3, `odo dev` starts the application on the cluster and automatically synchronizes the code.

> `odo dev` is a long-running process that blocks the terminal. When you enter `Ctrl+C`, `odo` v3 stops the process and automatically cleans up the component resources from the cluster.

Reviewer: ```suggestion
`odo dev` is a non-terminating process that blocks the terminal and runs until you enter `Ctrl+C`, after which it stops the process and automatically cleans up the component resources from the cluster.
```
